library/dash/src/main/java/com/google/android/exoplayer2/source/dash/manifest/DashManifestParser.java

```diff
@@ -517,7 +519,8 @@ protected RepresentationInfo parseRepresentation(
       } else if (XmlPullParserUtil.isStartTag(xpp, "SegmentList")) {
         segmentBase = parseSegmentList(xpp, (SegmentList) segmentBase);
       } else if (XmlPullParserUtil.isStartTag(xpp, "SegmentTemplate")) {
-        segmentBase = parseSegmentTemplate(xpp, (SegmentTemplate) segmentBase);
+        segmentBase = parseSegmentTemplate(xpp, (SegmentTemplate) segmentBase,
+            parentSupplementalProperties);
```

Comment:
Is it defined somewhere at what level the supplemental property will be defined? This seems to assume it'll be defined at the `AdaptationSet` level, but presumably `Period` and `Representation` levels could both be possible too?

Reply:
Level of the supplemental property is not defined. Have made changes to pass AdaptationSet and Representation supplemental properties to parse parseSegmentTemplate method.

Reply:
It seems non-ideal to push complexity into the client because something hasn't been defined in the DASH-IF doc. Can we assume it'll be defined at the same level as the `SegmentTemplate` for now, since this avoids having to pass around the supplemental properties from multiple layers.

I can open a request with the DASH-IF folks to get them to try and clarify how it's supposed to work in the spec.

Reply:
Makes sense. 
However, the asset provided by DASH-IF (https://testassets.dashif.org/#testvector/details/5a1c4b447cccbb15567e567b) has supplemental property at the AdaptationSet level and not SegmentTemplate level. 
Do you think it would make sense to assume that the supplemental property is specified at AdaptationSet level based on the example provided and pass around only the AdaptationSet supplemental properties? or request DASH-IF for clarification?

Reply:
Passing just the `AdaptationSet` ones seems OK for now (which I think is what you were doing before; sorry!). Thinking about this a bit more:

1. It wouldn't make sense for the property to be defined at the `Representation` level. Segments are supposed to be aligned across representations within an adaptation set, so the value would be the same in each one (i.e. may as well move the property up to the `AdaptationSet`)
1. I guess it's possible it could be defined at the `Period` level, but we've not seen any examples of that.

Reply:
Filed https://github.com/Dash-Industry-Forum/DASH-IF-IOP/issues/340

Reply:
Sounds good. Thanks